packages/toolpad-app/src/toolpadDataSources/client.tsx

```diff
@@ -1,23 +1,24 @@
+import * as _ from 'lodash-es';
 import functionSrc from './function/client';
 import postgres from './postgres/client';
 import rest from './rest/client';
 import { ClientDataSource } from '../types';
 import googleSheets from './googleSheets/client';
 import movies from './movies/client';
 import config from '../config';
+import { DEMO_DATASOURCES, PRODUCTION_DATASOURCES } from '../constants';
 
 type ClientDataSources = { [key: string]: ClientDataSource<any, any> | undefined };
 
-const clientDataSources: ClientDataSources = config.isDemo
-  ? {
-      movies,
-      rest,
-    }
-  : {
-      postgres,
-      function: functionSrc,
-      rest,
-      googleSheets,
-    };
+const clientDataSources: ClientDataSources = _.pick(
+  {
+    rest,
+    function: functionSrc,
+    postgres,
+    googleSheets,
+    movies,
+  },
+  [...PRODUCTION_DATASOURCES, ...(config.isDemo ? DEMO_DATASOURCES : [])],
```

Comment:
Is this correct? Shouldn't this just switch between the two lists?

```suggestion
  Array.from(config.isDemo ? DEMO_DATASOURCES : PRODUCTION_DATASOURCES),
```

Reply:
I need the list on the client side to be able to show which ones are only available when self-hosting...
Maybe I can get the full list from somewhere else though instead of defining it here.

Reply:
Perhaps we shouldn't do the filtering here then, and just export all of the datasources? Then do the filtering when we create the dropdown?

Reply:
this is outdated, i've already made it more like you suggested, lmk if it's cool